source/common/grpc/grpc_web_filter.cc

```diff
@@ -46,6 +46,7 @@ Http::FilterHeadersStatus GrpcWebFilter::decodeHeaders(Http::HeaderMap& headers,
   }
   is_grpc_web_request_ = true;
 
+  headers.removeContentLength();
```

Comment:
Please add a comment as to why this is being removed.

Reply:
Switching to chunked transfer encoding is what nginx does in this situation, fwiw

Reply:
@sayrer in grpc-web context the upstream must be http2, there is no chunked encoding, hence removing content length is enough.